### PR TITLE
[numpy][doc-fix] mean, transpose, stack, split, log2, rint and radians

### PR DESCRIPTION
**Pull request by mikemwx**
## Description ##
Add the docs for the above mentioned operators.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- three unary operators are registered to npi namespace, and wrapped in corresponding files
- `mean` is registered into npi namespace, and wrapped to have same signature as in numpy
- website building looks normal
- unittests for operators still pass

## Comments ##
Thank @reminisce , @haojin2  for reviewing

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -173,3 +173,112 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
         `axis` is not None or `a` is a 1-d array.
     """
     pass
+
+
+def _np_mean(a, axis=None, dtype=None, out=None, keepdims=None):
+    """
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: Add the real signature here: `mean(a, axis=None, dtype=None, out=None, keepdims=None)` and an empty line after it.

File: python/mxnet/_numpy_op_doc.py [hunk 2]
```diff
@@ -173,3 +173,112 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    Compute the arithmetic mean along the specified axis.
+    Returns the average of the array elements.
+    The average is taken over the flattened array by default, otherwise over the specified axis.
+
+    Parameters	
+    ----------
+    a : `NDArray`
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: Change all `NDArray` to `ndarray`.

File: python/mxnet/_numpy_op_doc.py [hunk 3]
```diff
@@ -173,3 +173,112 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    out : ndarray, optional
+        Alternate output array in which to place the result. The default is None; if provided,
+        it must have the same shape as the expected output, but the type will be cast if necessary. 
+    name : string, optional
+        Name of the resulting symbol
+
+    Returns
+    -------
+    p : ndarray
+        a with its axes permuted. A view is returned whenever possible.
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: view is not supported.

File: python/mxnet/ndarray/numpy/_op.py [hunk 4]
```diff
@@ -186,7 +186,7 @@ def stack(arrays, axis=0, out=None):
 
     Parameters
     ----------
-    arrays : sequence of array_like
+    arrays : sequence of ndarray
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: ndarray -> ndarrays

File: python/mxnet/_numpy_op_doc.py [hunk 5]
```diff
@@ -173,3 +173,112 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    array(0.55, dtype=float32)
+    >>> np.mean(a, dtype=np.float64)
+    array(0.55)
+    """
+    pass
+
+
+def _np_transpose(a, axes=None):
+    """
+    Permute the dimensions of an array.
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: Same as above, add the real signature here.

File: python/mxnet/_numpy_op_doc.py [hunk 6]
```diff
@@ -173,3 +173,112 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+
+    Parameters
+    ----------
+    a : NDArray
+        Input array.
+    axes : list of ints, optional
+        By default, reverse the dimensions,
+        otherwise permute the axes according to the values given.
+    out : ndarray, optional
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: `out` is not in the parameter list.

File: python/mxnet/ndarray/numpy/_op.py [hunk 7]
```diff
@@ -882,3 +927,112 @@ def sqrt(x, out=None, **kwargs):
     This function only supports input type of float.
     """
     return _unary_func_helper(x, _npi.sqrt, _np.sqrt, out=out, **kwargs)
+
+
+# pylint: disable=line-too-long
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: Do not add this. Make the line shorter if necessary.

File: python/mxnet/ndarray/numpy/_op.py [hunk 8]
```diff
@@ -882,3 +927,112 @@ def sqrt(x, out=None, **kwargs):
...
+        If not provided or None, a freshly-allocated array is returned.
+
+    Returns
+    -------
+    out : ndarray or scalar
+        Output array is same shape and type as x. This is a scalar if x is a scalar.
+
+    Notes
+    -----
+    This function differs from the original `numpy.rint
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: \`numpy.rint\`

Comment (mikemwx) [reply python/mxnet/ndarray/numpy/_op.py]: The other ` is on the next line after the web link which seems to be a necessary arrangement for the website building to work.

File: python/mxnet/_numpy_op_doc.py [hunk 9]
```diff
@@ -173,3 +173,107 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    a : `ndarray`
+        ndarray containing numbers whose mean is desired.
+    axis : None or int or tuple of ints, optional
+        Axis or axes along which the means are computed. The default is to compute the mean of the flattened array.
+        If this is a tuple of ints, a mean is performed over multiple axes, 
+        instead of a single axis or all the axes as before.
+    dtype : data-type, optional
+        Type to use in computing the mean. For integer inputs, the default is float32;
+        for floating point inputs, it is the same as the input dtype.
+    out : ndarray, optional
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Is `out` before `keepdims` or after in our version? I believe our `out` should always be the second last argument.

File: python/mxnet/ndarray/numpy/_op.py [hunk 10]
```diff
@@ -178,6 +179,68 @@ def minimum(x1, x2, out=None):
     return _ufunc_helper(x1, x2, _npi.minimum, _np.minimum, _npi.minimum_scalar, None, out)
 
 
+@set_module('mxnet.ndarray.numpy')
+def mean(a, axis=None, dtype=None, out=None, keepdims=False):  # pylint: disable=arguments-differ
```
Comment (gyshi) [inline python/mxnet/ndarray/numpy/_op.py]: the order of param  is right?

Comment (haojin2) [reply python/mxnet/ndarray/numpy/_op.py]: @mikemwx Please double-check and make sure the order of parameters is correct.

File: python/mxnet/numpy/multiarray.py [hunk 11]
```diff
@@ -924,7 +924,8 @@ def nanprod(self, *args, **kwargs):
 
     def mean(self, axis=None, dtype=None, out=None, keepdims=False):  # pylint: disable=arguments-differ
         """Returns the average of the array elements along given axis."""
-        return _mx_np_op.mean(self, axis=axis, dtype=dtype, keepdims=keepdims, out=out)
+        # return _mx_np_op.mean(self, axis=axis, dtype=dtype, keepdims=keepdims, out=out)
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: please remove dead code.

File: python/mxnet/numpy/multiarray.py [hunk 12]
```diff
@@ -1441,6 +1442,67 @@ def minimum(x1, x2, out=None):
         The minimum of x1 and x2, element-wise. This is a scalar if both x1 and x2 are scalars."""
     return _mx_nd_np.minimum(x1, x2, out=out)
 
+@set_module('mxnet.numpy')
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: One more blank line above (2 blank lines between functions)

File: python/mxnet/symbol/numpy/_symbol.py [hunk 13]
```diff
@@ -1020,29 +1020,117 @@ def mod(x1, x2, out=None):
 def power(x1, x2, out=None):
     return _ufunc_helper(x1, x2, _npi.power, _np.power, _npi.power_scalar, _npi.rpower_scalar, out)
 
+@set_module('mxnet.symbol.numpy')
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: One more blank line above.

File: python/mxnet/ndarray/numpy/_op.py [hunk 14]
```diff
@@ -17,7 +17,7 @@
 # under the License.
 
 """Namespace for numpy operators used in Gluon dispatched by F=ndarray."""
-
+# pylint: disable=too-many-lines
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: nit: Better use 1 more blank line above

File: python/mxnet/ndarray/numpy/_op.py (hunk 10 above)
Comment (mikemwx) [reply python/mxnet/ndarray/numpy/_op.py]: Yes, I have wrapped it

File: python/mxnet/_numpy_op_doc.py [hunk 15]
```diff
@@ -223,3 +223,46 @@ def _np_dot(a, b, out=None):
...
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.transpose.html>`_ in
+    the following way(s):
+
+    - only ndarray is accepted as valid input, python iterables are not supported
+
+    Examples
+    --------
+    >>> x = np.arange(4).reshape((2,2))
+    >>> x
+    array([[0., 1.],
```
Comment (tingying2020) [inline python/mxnet/_numpy_op_doc.py]: dtype=float32 is the default type and will not appear in the up to date version.

File: python/mxnet/ndarray/numpy/_op.py [hunk 16]
```diff
@@ -178,6 +180,68 @@ def minimum(x1, x2, out=None):
     return _ufunc_helper(x1, x2, _npi.minimum, _np.minimum, _npi.minimum_scalar, None, out)
 
 
+@set_module('mxnet.ndarray.numpy')
+def mean(a, axis=None, dtype=None, out=None, keepdims=False):  # pylint: disable=arguments-differ
+    """
+    mean(a, axis=None, dtype=None, out=None, keepdims=None)
```
Comment (tingying2020) [inline python/mxnet/ndarray/numpy/_op.py]: Here `out` should be the last parameter. Please double check it. And here are additional two parameters, `name` and `initial`.

Comment (tingying2020) [reply python/mxnet/ndarray/numpy/_op.py]: Please double check whether the argument should be named from the fourth.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 17]
```diff
@@ -1021,28 +1021,117 @@ def power(x1, x2, out=None):
...
+    a : `_Symbol`
+        _Symbol containing numbers whose mean is desired.
+    axis : None or int or tuple of ints, optional
+        Axis or axes along which the means are computed. The default is to compute the mean of the flattened array.
+        If this is a tuple of ints, a mean is performed over multiple axes,
+        instead of a single axis or all the axes as before.
+    dtype : data-type, optional
+        Type to use in computing the mean. For integer inputs, the default is float32;
+        for floating point inputs, it is the same as the input dtype.
+    out : _Symbol, optional
```
Comment (tingying2020) [inline python/mxnet/symbol/numpy/_symbol.py]: In _symbol.py, out is a dummy parameter to keep the consistency with the ndarray counterpart. Please double check.

File: python/mxnet/ndarray/numpy/_op.py (hunk 16 above)
Comment (mikemwx) [reply python/mxnet/ndarray/numpy/_op.py]: I have wrapped this function to only expose the above signature

File: python/mxnet/_numpy_op_doc.py (hunk 15 above)
Comment (mikemwx) [reply python/mxnet/_numpy_op_doc.py]: > dtype=float32 is the default type and will not appear in the up to date version.

Thank you for reminding. Will fix that.

**Pull request merged**